### ISSUE TITLE
Rescaling Back For Testing

### ISSUE THREAD

**Issue by amirgholami** [ellisdg/3DUnetCNN#132]
Hi Ellisdg

Many thanks for the nice code. Is there a utility function to transfer the results back to the uncropped original resolution of the prediction? For instance the train_isensee2017.py file uses whole brain but on 128^3 downsampled images after cropping it. I tried upsampling results and then zero padding based on the crop slices but there seems to be some issues with nii header files. I was wondering if this has already been implemented

Thanks,

**Comment by ellisdg** [comment]
I have [some code](https://github.com/ellisdg/3DUnetCNN/blob/master/unet3d/prediction.py#L102) that will get you part way there. Basically, you need to save the affine of the cropped and downsampled image(s). Then after you have the prediction data, you need to create a new image with NiBabel using the prediction data as well as the affine from the input images. This will give you the image in the correct orientation. To get the image back into the original sampling space, you can resample the predicted image with one of the uncropped input images using NiLearn's resample_to_img function.

**Comment by amirgholami** [comment]
Great many thanks. Basically I upsampled the logits to the uncropped size and manually corrected the affine. That seems to work. The resample_to_img will be a cleaner way to do it

**Comment by DecentMakeover** [comment]
@amirgholami Hey sorry to bother you but would it be possible for you to share the code?Thanks
